core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala

```diff
@@ -777,6 +779,69 @@ class WorkspaceServiceSpec extends AnyFlatSpec with ScalatestRouteTest with Matc
 
   }
 
+  it should "handle 404s from Sam when deleting a workspace" in withTestDataServices { services =>
+    //check that the workspace to be deleted exists
+    assertWorkspaceResult(Option(testData.workspaceNoSubmissions)) {
+      runAndWait(workspaceQuery.findByName(testData.wsName3))
+    }
+
+    when(
+      services.samDAO.deleteResource(ArgumentMatchers.eq(SamResourceTypeNames.workspace), any[String], any[UserInfo])
```

Comment:
⛏️ Nitpick: consider matching against `testData.workspaceNoSubmissions.workspaceId` rather than the looser `any[String]` to avoid any false matches.

Reply:
Sure, I updated workspace deletion calls. I didn't change the workflowCollection ones though because I didn't want to have to figure out the workflowCollection id (and I didn't want to risk the test trivially passing because I had the wrong ID).